src/runtime/policy/ingress-validation.ts

```diff
@@ -162,6 +162,40 @@ export class IngressValidation {
     return this.policies.some(policy => policy.targets.some(
       target => target.schemaName === type.getEntitySchema().name));
   }
+
+  // Get the max readable type for `typeName` according to the given `policies`.
+  static getMaxReadType(typeName: string, policies: Policy[]): Type|null {
+    const fields = {};
+    let type = null;
+    for (const policy of policies) {
+      for (const target of policy.targets) {
+        if (typeName === target.schemaName) {
+          type = target.type;
+          IngressValidation.mergeFields(fields, target.getRestrictedFields());
+        }
+      }
+    }
+    return type ? EntityType.make([typeName], fields, type.getEntitySchema()) : null;
+  }
+
+  private static mergeFields(fields: {}, newFields: {}) {
+    for (const newFieldName of Object.keys(newFields)) {
+      if (fields[newFieldName]) {
+        IngressValidation.mergeField(fields[newFieldName], newFields[newFieldName]);
+      } else {
+        fields[newFieldName] = newFields[newFieldName];
+      }
+    }
+  }
+
+  private static mergeField(field: FieldType, newField: FieldType) {
+    assert(field.kind === newField.kind);
```

Comment:
Do you want to add a meaningful message here? e.g. `Field '${typeName}.${name}' has incompatible type across policies`, or something like that.